### DISCUSSION TITLE
Trapdoors and villagers

### DISCUSSION THREAD

**Discussion by UsainSrht** [PurpurMC/Purpur#594] [__Dismissed__] [answered]
https://user-images.githubusercontent.com/59025999/129879641-e743f8ff-0e3c-442f-bf5f-17dccf0b7d01.mp4

I've enabled lobotimizing villagers but villagers can pathfing through trapdoors may be new feature can be added for disabling this.

**Comment by UsainSrht** [comment, marked as answer]
idk why but cant play video on github so i uploaded it to streamable https://streamable.com/ptbgw3

**Reply by CenCent** [reply]
You can remove their AI by putting 2 or 3 (I don't remember) layers of carpet under the villagers. It makes them think they are falling as carpets don't count as walkable block :)

**Reply by UsainSrht** [reply]
Could be but telling every player to do that is not an option.
